applications/CompressiblePotentialFlowApplication/custom_operations/potential_to_compressible_navier_stokes_operation.cpp

```diff
@@ -0,0 +1,117 @@
+    auto& mDestinationModelPart = mpModel->GetModelPart(destination_model_part_name);
+
+    const int n_orig_nodes = mOriginModelPart.NumberOfNodes();
+    const int n_dest_nodes = mDestinationModelPart.NumberOfNodes();
+
+    // Check number of nodes
+    KRATOS_ERROR_IF_NOT(n_orig_nodes == n_dest_nodes)
+        << "Origin and destination model parts have different number of nodes."
+        << "\n\t- Number of origin nodes: " << n_orig_nodes
+        << "\n\t- Number of destination nodes: " << n_dest_nodes << std::endl;
+
+    // Getting the required variables
+    const double& gamma = mOriginModelPart.GetProcessInfo().GetValue(HEAT_CAPACITY_RATIO);
+    const double& sound_velocity = mOriginModelPart.GetProcessInfo().GetValue(SOUND_VELOCITY);
+    const double& free_stream_rho = mOriginModelPart.GetProcessInfo().GetValue(FREE_STREAM_DENSITY);
+    const double& free_stream_mach = mOriginModelPart.GetProcessInfo().GetValue(FREE_STREAM_MACH);
+    const double& specific_heat = (std::pow(sound_velocity,2) / (gamma * reference_temperature)) / (gamma - 1.0);
+    
+    // Calculate de nodal velocities
+    if (compute_nodal_velocities){
+        // Construct the ComputeNodalPotentialFlowVelocityProcess
+        const std::vector<std::string> variable_array = {"VELOCITY"};
+        ComputeNodalValueProcess ComputeNodalValueProcess(mOriginModelPart, variable_array);
+
+        // Execute the ComputeNodalPotentialFlowVelocityProcess
+        ComputeNodalValueProcess.Execute();}
+
+    // Transfer the potential flow values as initial condition for the compressible problem
+    IndexPartition<std::size_t>(n_orig_nodes).for_each([&](std::size_t index)
+                                                       {
+        auto it_dest_node = mDestinationModelPart.NodesBegin() + index;
+        const auto it_orig_node = mOriginModelPart.NodesBegin() + index;
+
+        // Getting potential flow velocities
+        const auto &velocity = it_orig_node->GetValue(VELOCITY);
+        
+        // Calculate the conservative variables
+        const double& velocity_norm_2 = velocity[0] * velocity[0] + velocity[1] * velocity[1];
+        const double& velocity_norm = std::pow(velocity_norm_2,0.5);
+        const double& mach = velocity_norm / sound_velocity;
+        const double& num = 1.0 + 0.5 * (gamma - 1.0) * std::pow(free_stream_mach,2);
+        const double& den = 1.0 + 0.5 * (gamma - 1.0) * std::pow(mach,2);
+        const double& density = free_stream_rho * std::pow((num / den),(1.0 / (gamma - 1.0)));
+        const double& energy = specific_heat * reference_temperature + 0.5 * velocity_norm_2;
+        
+        // Setting the Navier Stokes initial condition
+        it_dest_node->FastGetSolutionStepValue(DENSITY, 0) = density; 
+        it_dest_node->FastGetSolutionStepValue(MOMENTUM, 0) = density*velocity;
+        it_dest_node->FastGetSolutionStepValue(TOTAL_ENERGY, 0) = density*energy; });
```

Comment:
```suggestion
        it_dest_node->FastGetSolutionStepValue(DENSITY) = density; 
        it_dest_node->FastGetSolutionStepValue(MOMENTUM) = density*velocity;
        it_dest_node->FastGetSolutionStepValue(TOTAL_ENERGY) = density*energy;
    });
```
Besides this minor, I think that we should set the values in all the buffer positions. In other words I'd initialize the history as well.